GitCommands/Git/GitModule.cs

```diff
@@ -2611,41 +2611,41 @@ public bool IsDirtyDir()
         }
 
         [CanBeNull]
-        public Patch GetCurrentChanges(string fileName, [CanBeNull] string oldFileName, bool staged, string extraDiffArguments, Encoding encoding, bool noLocks = false)
+        public async Task<Patch> GetCurrentChangesAsync(string fileName, [CanBeNull] string oldFileName, bool staged, string extraDiffArguments, Encoding encoding, bool noLocks = false)
         {
-            var output = _gitExecutable.GetOutput(GetCurrentChangesCmd(fileName, oldFileName, staged, extraDiffArguments, encoding, noLocks),
-                outputEncoding: LosslessEncoding);
+            var output = await _gitExecutable.GetOutputAsync(GetCurrentChangesCmd(fileName, oldFileName, staged, extraDiffArguments, encoding, noLocks),
+                outputEncoding: LosslessEncoding).ConfigureAwait(false);
 
             var patches = PatchProcessor.CreatePatchesFromString(output, encoding).ToList();
 
             return GetPatch(patches, fileName, oldFileName);
         }
 
         [CanBeNull]
-        private string GetFileContents(string path)
+        private async Task<string> GetFileContentsAsync(string path)
         {
             var args = new GitArgumentBuilder("show") { $"HEAD:{path.ToPosixPath().Quote()}" };
-            var result = _gitExecutable.Execute(args);
+            var result = await _gitExecutable.ExecuteAsync(args).ConfigureAwaitRunInline();
```

Comment:
Could you please explain how you chose when to use `ConfigureAwaitRunInline()` and when `ConfigureAwait(false)` calls?

I've found https://github.com/Microsoft/vs-threading/pull/292 (via https://github.com/dotnet/project-system/pull/3662), however it is not clear to me why we aren't using one or the other.

Reply:
I use `ConfigureAwait(false)` when I don't want the continuation to be executed on the main thread.
I use `ConfigureAwaitRunInline` when I know it is safe to continue on the thread that completed the awaited task (regarding both: accessing variables and continuation's duration time).